Replace spacing near scroll views with overscroll (COMMUNTIY)

fynngodau opened the pull request:
As previously discussed in https://github.com/corona-warn-app/cwa-app-android/pull/1882#discussion_r543602525, this PR removes white space near scroll views, mostly at the bottom, and instead adds that space inside the scroll view (so it scrolls with the content). This makes the app feel a lot better.

Here you can see some examples:

Before | After
---|---
![Screenshot_1618984437](https://user-images.githubusercontent.com/16943720/115506681-6ee34f80-a27b-11eb-9a8b-dbbb7517a016.png)| ![Screenshot_1618984392](https://user-images.githubusercontent.com/16943720/115506703-786cb780-a27b-11eb-965b-2f93e9e14b19.png)
![Screenshot_1618986546](https://user-images.githubusercontent.com/16943720/115506906-b9fd6280-a27b-11eb-9ac4-1124849cbd37.png) | ![Screenshot_1618986818](https://user-images.githubusercontent.com/16943720/115507413-62132b80-a27c-11eb-95a0-e87a4dd0e738.png)
![Screenshot_1618987264](https://user-images.githubusercontent.com/16943720/115508188-52e0ad80-a27d-11eb-9ca2-dae64330d83e.png) | ![Screenshot_1618987190](https://user-images.githubusercontent.com/16943720/115508205-55db9e00-a27d-11eb-8e2e-831e9ac5c263.png)


These views have been fixed that way:

* App Information
* About the App
* Data Privacy
* Legal Notices
* Technical Hotline
* Statistics explanation
* Edit Places
* Edit People
* Settings
* Notifications
* Prioritized Background Activity
* Share Data
* Check In (onboarding / information)

The error report screen has also been fixed to remove the white space between the elevated, sticky content and the scrollable content. Note that here, I changed the total bottom padding of the scrollable content from `20dp` to `24dp` for consistency within the layout.

Additionally, I fixed the button in the check in layout to have the same margin to top and bottom as the Open Journal button in the Contact Journal onboarding / information screen.

---
Internal Tracking ID: [EXPOSUREAPP-6605](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6605)

fynngodau committed:
Remove space below scrollers and add overscroll in app info screens

Note that this changes the total padding of the content in the debuglog
fragment from 20dp to 24dp for consistency within the layout.

fynngodau committed:
Remove spacing below scroll view and add overscroll to stats explanation

fynngodau committed:
Fix scrollbars

fynngodau committed:
Fix contact diary (edit places & locations) scroll behavior

dsarkar commented:
@fynngodau Thanks. Internal Tracking ID: [EXPOSUREAPP-6605](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6605)


---
Corona-Warn-App Open Source Team

harambasicluka committed:
Merge branch 'release/2.1.x' into fix/overscroll

harambasicluka committed:
Merge branch 'release/2.1.x' into fix/overscroll

fynngodau committed:
Remove space below scrollers and add overscroll in settings screens

fynngodau commented:
I added fixes for the following layouts, which I initially overlooked:

* Settings
* Notifications (notification settings)
* Prioritized Background Activity (battery optimization settings)
* Share Data (ppa settings)
* Detailed Information about Data Sharing

fynngodau committed:
Undo changes to PPA more info fragment

This partially reverts commit 85e4650e12e27999a01a293695121cd324a7123f
to avoid conflicts with #2945.

fynngodau commented:
I undid the change to the Detailed Information about Data Sharing screen to avoid conflict with #2945.

fynngodau committed:
Fix Check In onboarding

Also changes margin of button to match that of contact journal
onboarding

fynngodau commented:
One more screen:

* Check In onbaording

Also fixes the margin of the Check In onboarding Agree button to match that of the Open Journal button in the Contact Journal onboarding.

harambasicluka committed:
Merge branch 'release/2.1.x' into fix/overscroll

harambasicluka committed:
Merge branch 'release/2.1.x' into fix/overscroll

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_information_about.xml:
```diff
@@ -24,7 +24,10 @@
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
I would align the bottom padding on this kind of fragments, I would go with `@dimen/spacing_normal` as it is here: https://github.com/corona-warn-app/cwa-app-android/pull/2884/files#diff-99108bde996a7dd32ce38fb5594a027efd59819f2b88d116dcc7b8f3eaced7ebR30

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_information_contact.xml:
```diff
@@ -24,7 +24,10 @@
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_information_legal.xml:
```diff
@@ -31,9 +31,11 @@
-            android:fillViewport="true"
             android:focusable="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_information_privacy.xml:
```diff
@@ -30,7 +30,9 @@
             android:layout_height="match_parent"
             android:fillViewport="true"
-            android:paddingBottom="@dimen/guideline_bottom">
+            android:paddingBottom="@dimen/guideline_bottom"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_information_technical.xml:
```diff
@@ -30,7 +30,9 @@
             android:layout_height="match_parent"
             android:fillViewport="true"
-            android:paddingBottom="@dimen/guideline_bottom">
+            android:paddingBottom="@dimen/guideline_bottom"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_settings.xml:
```diff
@@ -43,7 +43,10 @@
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml:
```diff
@@ -31,7 +31,10 @@
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_settings_notifications.xml:
```diff
@@ -32,7 +32,10 @@
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:
```diff
@@ -27,7 +27,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
-            app:layout_constraintBottom_toTopOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml:
```diff
@@ -93,7 +93,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
-            app:layout_constraintBottom_toBottomOf="@id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Same as above. But there seems to be already more space.

fynngodau replied:
The answer is here:

https://github.com/corona-warn-app/cwa-app-android/blob/e1ec946e09600826d16ea389aa8dd96f7c959129/Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml#L431

That's an odd margin… we should remove that before increasing the scroll view's padding.

harambasicluka commented on Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml:
```diff
@@ -31,7 +32,9 @@
         android:layout_width="match_parent"
         android:layout_height="0dp"
-        android:layout_marginBottom="20dp"
+        android:paddingBottom="@dimen/spacing_small"
```
Seems fine, as the floating button is below.

harambasicluka reviewed (changes requested):
Overall looks pretty good, but I would like to take this chance to align all bottom paddings :)

What do you think?

fynngodau committed:
Uniform bottom paddings

fynngodau commented:
@harambasicluka Agreed – I have pushed a change updating all the places you pointed out.

chiljamgossow reviewed (commented):
Very nice! Thanks for fixing.

harambasicluka committed:
Merge branch 'release/2.1.x' into fix/overscroll

harambasicluka committed:
Merge branch 'release/2.1.x' into fix/overscroll
